File: Android 07 - Gradle and External libraries/ExternalLibraryExercise/app/src/main/java/com/programming/android/sdu/externallibraryexercise/RandomPicturesAdapter.java
package com.programming.android.sdu.externallibraryexercise;

import android.content.Context;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by grzegorzbaczek on 07/03/2018.
 */

public class RandomPicturesAdapter extends RecyclerView.Adapter<RandomPicturesAdapter.ViewHolder> {
    private Map<Integer, String> mDataset = new HashMap<>();
    private Context context;

    public  class ViewHolder extends RecyclerView.ViewHolder {
        public ImageView imageView;
        public ViewHolder(View v) {
            super(v);
            imageView = v.findViewById(R.id.ivCell);
        }
    }

    public RandomPicturesAdapter(Context context){
        this.context = context;
    }

    @Override
    public RandomPicturesAdapter.ViewHolder onCreateViewHolder(ViewGroup parent,
                                                               int viewType) {
        View v =  LayoutInflater.from(parent.getContext())
                .inflate(R.layout.cell_layout, parent, false);
        ViewHolder vh = new ViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        String urlToCall = String.format("https://picsum.photos/500/500?image=%d", position);
        Glide.with(context).clear(holder.imageView);
        Glide.with(context).load(urlToCall).into(holder.imageView);
    }

    @Override
    public int getItemCount() {
        return 50;
    }
}

File: Android 07 - Gradle and External libraries/ExternalLibraryExercise/app/src/main/java/com/programming/android/sdu/externallibraryexercise/RandomPicturesAdapter.kt
package com.programming.android.sdu.externallibraryexercise

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide

/**
 * Created by grzegorzbaczek on 07/03/2018.
 */
class RandomPicturesAdapter(private val context: Context) : RecyclerView.Adapter<RandomPicturesAdapter.ViewHolder>() {

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var imageView: ImageView = v.findViewById(R.id.ivCell)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.cell_layout, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val urlToCall = String.format("https://picsum.photos/500/500?image=%d", position)
        Glide.with(context).clear(holder.imageView)
        Glide.with(context).load(urlToCall).into(holder.imageView)
    }

    override fun getItemCount(): Int {
        return 50
    }

}